Failing on windows 11 at cargo build --release

Hi, I am trying to figure out the problem, help will be great.

Failing on windows 11 at cargo build --release

Error message:

`Caused by:
  process didn't exit successfully: `C:\Users\touhidul.alam\Desktop\llama\llama-rs\target\release\build\ggml-raw-2925d7a622a7c725\build-script-build` (exit code: 1)
  --- stdout
  TARGET = Some("x86_64-pc-windows-gnu")
  OPT_LEVEL = Some("3")
  HOST = Some("x86_64-pc-windows-gnu")
  cargo:rerun-if-env-changed=CC_x86_64-pc-windows-gnu
  CC_x86_64-pc-windows-gnu = None
  cargo:rerun-if-env-changed=CC_x86_64_pc_windows_gnu
  CC_x86_64_pc_windows_gnu = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64-pc-windows-gnu
  CFLAGS_x86_64-pc-windows-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_pc_windows_gnu
  CFLAGS_x86_64_pc_windows_gnu = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "gcc.exe" "-O3" "-ffunction-sections" "-fdata-sections" "-m64" "-I" "include" "-Wall" "-Wextra" "/arch:AVX2" "-DNDEBUG" "-o" "C:\\Users\\touhidul.alam\\Desktop\\llama\\llama-rs\\target\\release\\build\\ggml-raw-6f69f4f538ccd1b4\\out\\ggml/ggml.o" "-c" "ggml/ggml.c"
  cargo:warning=ggml/ggml.c: In function 'pthread_create':
  cargo:warning=ggml/ggml.c:53:49: warning: unused parameter 'unused' [-Wunused-parameter]
  cargo:warning=   53 | static int pthread_create(pthread_t* out, void* unused, thread_ret_t(*func)(void*), void* arg) {
  cargo:warning=      |                                           ~~~~~~^~~~~~
  cargo:warning=ggml/ggml.c: In function 'pthread_join':
  cargo:warning=ggml/ggml.c:64:49: warning: unused parameter 'unused' [-Wunused-parameter]
  cargo:warning=   64 | static int pthread_join(pthread_t thread, void* unused) {
  cargo:warning=      |                                           ~~~~~~^~~~~~
  cargo:warning=gcc.exe: warning: /arch:AVX2: linker input file unused because linking not done
  cargo:warning=gcc.exe: error: /arch:AVX2: linker input file not found: No such file or directory
  exit code: 1

  --- stderr


  error occurred: Command "gcc.exe" "-O3" "-ffunction-sections" "-fdata-sections" "-m64" "-I" "include" "-Wall" "-Wextra" "/arch:AVX2" "-DNDEBUG" "-o" "C:\\Users\\touhidul.alam\\Desktop\\llama\\llama-rs\\target\\release\\build\\ggml-raw-6f69f4f538ccd1b4\\out\\ggml/ggml.o" "-c" "ggml/ggml.c" with args "gcc.exe" did not execute successfully (status code exit code: 1).`

--- a/ggml-raw/build.rs
+++ b/ggml-raw/build.rs
@@ -15,34 +15,33 @@ fn main() {
     // This is a very basic heuristic for applying compile flags.
     // Feel free to update this to fit your operating system.
     let target_arch = env::var("CARGO_CFG_TARGET_ARCH").unwrap();
-    let target_os = env::var("CARGO_CFG_TARGET_OS").unwrap();
     let is_release = env::var("PROFILE").unwrap() == "release";
+    let compiler = build.get_compiler();
 
     match target_arch.as_str() {
         "x86" | "x86_64" => {
             let features = x86::Features::get();
 
-            match target_os.as_str() {
-                "freebsd" | "haiku" | "ios" | "macos" | "linux" => {
-                    build.flag("-pthread");
+            if compiler.is_like_clang() || compiler.is_like_gnu() {
+                build.flag("-pthread");
 
-                    if features.avx {
-                        build.flag("-mavx");
-                    }
-                    if features.avx2 {
-                        build.flag("-mavx2");
-                    }
-                    if features.fma {
-                        build.flag("-mfma");
-                    }
-                    if features.f16c {
-                        build.flag("-mf16c");
-                    }
-                    if features.sse3 {
-                        build.flag("-msse3");
-                    }
+                if features.avx {
+                    build.flag("-mavx");
                 }
-                "windows" => match (features.avx2, features.avx) {
+                if features.avx2 {
+                    build.flag("-mavx2");
+                }
+                if features.fma {
+                    build.flag("-mfma");
+                }
+                if features.f16c {
+                    build.flag("-mf16c");
+                }
+                if features.sse3 {
+                    build.flag("-msse3");
+                }
+            } else if compiler.is_like_msvc() {
+                match (features.avx2, features.avx) {
                     (true, _) => {
                         build.flag("/arch:AVX2");
                     }
@@ -50,8 +49,7 @@ fn main() {
                         build.flag("/arch:AVX");
                     }
                     _ => {}
-                },
-                _ => {}
+                }
             }
         }
         _ => {}

